icu4j/main/classes/core/src/com/ibm/icu/impl/units/MeasureUnitImpl.java

```diff
@@ -372,34 +370,32 @@ public int getValue() {
     public static class UnitsParser {
         // This used only to not build the trie each time we use the parser
         private volatile static CharsTrie savedTrie = null;
-        private final String[] simpleUnits;
+
         // This trie used in the parsing operation.
-        private CharsTrie trie;
+        private final CharsTrie trie;
+        private final String fSource;
         // Tracks parser progress: the offset into fSource.
         private int fIndex = 0;
         // Set to true when we've seen a "-per-" or a "per-", after which all units
         // are in the denominator. Until we find an "-and-", at which point the
         // identifier is invalid pending TODO(CLDR-13700).
         private boolean fAfterPer = false;
-        private String fSource;
         // If an "-and-" was parsed prior to finding the "single
         //     * unit", sawAnd is set to true. If not, it is left as is.
         private boolean fSawAnd = false;
 
         private UnitsParser(String identifier) {
-            this.simpleUnits = UnitsData.getSimpleUnits();
             this.fSource = identifier;
 
-            if (UnitsParser.savedTrie != null) {
-                try {
-                    this.trie = UnitsParser.savedTrie.clone();
-                } catch (CloneNotSupportedException e) {
-                    throw new ICUCloneNotSupportedException();
-                }
-                return;
+            try {
+                this.trie = UnitsParser.savedTrie.clone();
+            } catch (CloneNotSupportedException e) {
+                throw new ICUCloneNotSupportedException();
             }
+        }
 
-            // Building the trie.
+        static {
```

Comment:
I think `static {}` blocks run when the class gets loaded.  Is that more often than we want?  If someone isn't doing unit formatting, we don't want to have them wait for this trie to be built and take up their memory.

Reply:
s/unit formatting/unit identifier parsing/ ?

I think this initialization code is needed whenever this class (UnitsParser) is used?

So the question is just whether nested classes initialize in the way I think they do? Instantiating MeasureUnitImpl but not parsing identifiers (not calling MeasureUnitImpl.forIdentifier) should result in this not being run? We could test with a quick "system.out.println" in a unit test that does this?

Reply:
I agree with @hugovdm